# Prueba de Hipótesis Unimuestral (Prueba Z)

## Planteamiento

Se desea determinar si el precio promedio de las acciones registradas en el dataset es igual a **2.50 USD**, valor tomado como referencia debido a que representa un precio representativo dentro del rango donde se concentra la mayor parte de las observaciones.

El parámetro de interés es la media poblacional ($\mu$) del precio de las acciones registradas en el dataset.

Como el tamaño de la muestra es considerablemente grande (\(n > 30\)), se aplica una **prueba Z para una muestra**, sustentada en el Teorema Central del Límite.



## Hipótesis

Hipótesis nula:

$$
H_0:\mu = 2.50
$$

Hipótesis alternativa:

$$
H_1:\mu \neq 2.50
$$



## Nivel de significancia

Se establece un nivel de significancia de

$$
\alpha = 0.05
$$



## Estadístico de prueba

El estadístico utilizado es

$$
Z=\frac{\bar{x}-\mu_0}{\sigma/\sqrt{n}}
$$

Debido a que la desviación estándar poblacional es desconocida, se utiliza la desviación estándar muestral como estimación:

$$
Z=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}
$$

donde

- $\bar{x}$ : media muestral.
- $\mu_0$ : media hipotética.
- $s$ : desviación estándar muestral.
- $n$ : tamaño de la muestra.



## Regla de decisión

El valor-p se obtiene mediante la distribución normal estándar.

Si

$$
p<\alpha
$$

se rechaza la hipótesis nula.

En caso contrario,

$$
p\ge\alpha
$$

no se rechaza la hipótesis nula.

In [ ]:
# ============================================================
# PRUEBA DE HIPÓTESIS UNIMUESTRAL (PRUEBA Z)
# ============================================================

import numpy as np
import pandas as pd
from scipy import stats

# ------------------------------------------------------------
# Cargar el conjunto de datos
# ------------------------------------------------------------

columnas = [
    "Fecha", "Tipo", "Emisor", "Cantidad",
    "Valor_Nominal", "Precio",
    "Valor_Nominal_Total", "Monto_Efectivo",
    "Casa_Origen", "Casa_Destino", "Bolsa"
]

df = pd.read_csv(
    "/content/BVG_Acciones_export.csv",
    names=columnas,
    skiprows=1
)

# Convertir la columna Precio a formato numérico
df["Precio"] = pd.to_numeric(df["Precio"], errors="coerce")

# Eliminar valores faltantes
precios = df["Precio"].dropna()

# ------------------------------------------------------------
# Parámetros de la prueba
# ------------------------------------------------------------

mu0 = 2.50
alpha = 0.05

# ------------------------------------------------------------
# Estadísticos muestrales
# ------------------------------------------------------------

n = len(precios)
media = precios.mean()
desviacion = precios.std(ddof=1)
error_estandar = desviacion / np.sqrt(n)

# ------------------------------------------------------------
# Estadístico Z
# ------------------------------------------------------------

# La desviación estándar poblacional es desconocida.
# Debido al gran tamaño de la muestra (n > 30),
# se aplica una prueba Z utilizando la desviación
# estándar muestral como estimación.

z = (media - mu0) / error_estandar

# Valor-p bilateral usando scipy.stats
p_value = 2 * stats.norm.sf(abs(z))

# ------------------------------------------------------------
# Resultados
# ------------------------------------------------------------

print("="*60)
print("PRUEBA Z PARA UNA MUESTRA")
print("="*60)

print(f"Tamaño de la muestra          : {n}")
print(f"Media muestral               : {media:.4f}")
print(f"Desviación estándar          : {desviacion:.4f}")
print(f"Error estándar               : {error_estandar:.6f}")
print(f"Media hipotética (μ₀)        : {mu0:.2f}")
print(f"Nivel de significancia (α)   : {alpha}")

print("\n" + "-"*60)

print(f"Estadístico Z                : {z:.4f}")
print(f"Valor-p                      : {p_value:.6f}")

print("-"*60)

# ------------------------------------------------------------
# Decisión estadística
# ------------------------------------------------------------

if p_value < alpha:
    print("Decisión:")
    print("Se rechaza la hipótesis nula (H₀).")
    print("Existe evidencia estadística suficiente para concluir")
    print("que el precio promedio es diferente de 2.50 USD.")
else:
    print("Decisión:")
    print("No se rechaza la hipótesis nula (H₀).")
    print("No existe evidencia estadística suficiente para concluir")
    print("que el precio promedio es diferente de 2.50 USD.")

PRUEBA Z PARA UNA MUESTRA
Tamaño de la muestra          : 29131
Media muestral               : 42.6439
Desviación estándar          : 240.8388
Error estándar               : 1.411071
Media hipotética (μ₀)        : 2.50
Nivel de significancia (α)   : 0.05

------------------------------------------------------------
Estadístico Z                : 28.4493
Valor-p                      : 0.000000
------------------------------------------------------------
Decisión:
Se rechaza la hipótesis nula (H₀).
Existe evidencia estadística suficiente para concluir
que el precio promedio es diferente de 2.50 USD.


# Interpretación de Resultados

La prueba Z para una muestra permitió contrastar la hipótesis sobre el precio promedio de las acciones registradas en el dataset.

Se obtuvo un estadístico de prueba de:

$$
Z = 28.4493
$$

y un valor-p de:

$$
p = 0.000000
$$

Con un nivel de significancia de:

$$
\alpha = 0.05
$$

Dado que

$$
p < \alpha
$$

se **rechaza la hipótesis nula ($H_0$)**.

Por lo tanto, existe evidencia estadística suficiente para concluir que el **precio promedio de las acciones registradas en el dataset es significativamente diferente de 2.50 USD**.

En este caso, la media muestral obtenida fue de **42.6439 USD**, valor considerablemente superior al valor hipotético planteado, lo que confirma que el promedio de los precios de las acciones difiere del valor de referencia establecido para la prueba.

# Comparación de Grupos mediante ANOVA de un Factor

## Planteamiento

Se desea determinar si existen diferencias significativas en el **precio promedio de las acciones** entre distintos meses del año registrados en el dataset.

Para ello, las observaciones se dividen en cuatro grupos de acuerdo con el mes de la transacción:

- Enero
- Abril
- Julio
- Octubre

Debido a que se comparan **más de dos grupos independientes**, se emplea un **Análisis de Varianza (ANOVA) de un factor**.



## Hipótesis

Hipótesis nula:

$$
H_0:\mu_1=\mu_2=\mu_3=\mu_4
$$

Es decir, las medias poblacionales de los cuatro grupos son iguales.

Hipótesis alternativa:

$$
H_1:\exists\, i,j \text{ tal que } \mu_i \neq \mu_j
$$

Es decir, al menos una de las medias poblacionales es diferente.



## Nivel de significancia

Se establece:

$$
\alpha=0.05
$$



## Estadístico de prueba

El estadístico utilizado por el ANOVA es

$$
F=\frac{MS_{Entre}}{MS_{Dentro}}
$$

donde

- $MS_{Entre}$ representa la variabilidad entre los grupos.
- $MS_{Dentro}$ representa la variabilidad dentro de los grupos.



## Regla de decisión

Si

$$
p<\alpha
$$

se rechaza la hipótesis nula.

En caso contrario,

$$
p\geq\alpha
$$

no se rechaza la hipótesis nula.

Si el ANOVA resulta significativo, se aplicará posteriormente la **prueba Post-Hoc de Tukey** para identificar cuáles grupos presentan diferencias significativas.

In [ ]:
# ============================================================
# ANÁLISIS DE VARIANZA (ANOVA DE UN FACTOR)
# ============================================================

import pandas as pd
from scipy import stats

# ------------------------------------------------------------
# Cargar el dataset
# ------------------------------------------------------------

columnas = [
    "Fecha", "Tipo", "Emisor", "Cantidad",
    "Valor_Nominal", "Precio",
    "Valor_Nominal_Total", "Monto_Efectivo",
    "Casa_Origen", "Casa_Destino", "Bolsa"
]

df = pd.read_csv(
    "/content/BVG_Acciones_export.csv",
    names=columnas,
    skiprows=1
)

# ------------------------------------------------------------
# Preparación de los datos
# ------------------------------------------------------------

df["Fecha"] = pd.to_datetime(df["Fecha"], errors="coerce")
df["Precio"] = pd.to_numeric(df["Precio"], errors="coerce")

df = df.dropna(subset=["Fecha", "Precio"])

# Extraer el mes
df["Mes"] = df["Fecha"].dt.month

# ------------------------------------------------------------
# Crear los grupos
# ------------------------------------------------------------

enero = df[df["Mes"] == 1]["Precio"]
abril = df[df["Mes"] == 4]["Precio"]
julio = df[df["Mes"] == 7]["Precio"]
octubre = df[df["Mes"] == 10]["Precio"]

# ------------------------------------------------------------
# Ejecutar ANOVA
# ------------------------------------------------------------

F, p_value = stats.f_oneway(
    enero,
    abril,
    julio,
    octubre
)

# ------------------------------------------------------------
# Resultados
# ------------------------------------------------------------

print("="*60)
print("ANOVA DE UN FACTOR")
print("="*60)

print(f"Enero   : {len(enero)} observaciones")
print(f"Abril   : {len(abril)} observaciones")
print(f"Julio   : {len(julio)} observaciones")
print(f"Octubre : {len(octubre)} observaciones")

print("\n")

print(f"Estadístico F : {F:.4f}")
print(f"Valor-p       : {p_value:.6f}")

print("\n")

alpha = 0.05

if p_value < alpha:
    print("Decisión:")
    print("Se rechaza la hipótesis nula.")
    print("Existen diferencias significativas entre las medias.")
else:
    print("Decisión:")
    print("No se rechaza la hipótesis nula.")
    print("No existen diferencias significativas entre las medias.")

ANOVA DE UN FACTOR
Enero   : 2639 observaciones
Abril   : 2261 observaciones
Julio   : 2402 observaciones
Octubre : 2256 observaciones


Estadístico F : 4.9412
Valor-p       : 0.001984


Decisión:
Se rechaza la hipótesis nula.
Existen diferencias significativas entre las medias.


In [ ]:
# ============================================================
# PRUEBA POST-HOC DE TUKEY
# ============================================================

from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Mantener únicamente los meses analizados
df_tukey = df[df["Mes"].isin([1, 4, 7, 10])].copy()

# Cambiar números por nombres
nombres = {
    1: "Enero",
    4: "Abril",
    7: "Julio",
    10: "Octubre"
}

df_tukey["Mes"] = df_tukey["Mes"].map(nombres)

# Ejecutar Tukey
tukey = pairwise_tukeyhsd(
    endog=df_tukey["Precio"],
    groups=df_tukey["Mes"],
    alpha=0.05
)

print(tukey)

 Multiple Comparison of Means - Tukey HSD, FWER=0.05  
group1  group2 meandiff p-adj   lower    upper  reject
------------------------------------------------------
 Abril   Enero  21.0549 0.0013   6.3768  35.733   True
 Abril   Julio   16.522 0.0242   1.5135 31.5306   True
 Abril Octubre  14.8994 0.0582  -0.3428 30.1416  False
 Enero   Julio  -4.5329 0.8515 -18.9771  9.9114  False
 Enero Octubre  -6.1555 0.7037 -20.8424  8.5313  False
 Julio Octubre  -1.6227 0.9925 -16.6398 13.3944  False
------------------------------------------------------


# Interpretación de Resultados

El análisis de varianza (ANOVA) de un factor permitió comparar el **precio promedio de las acciones** entre los meses de **enero, abril, julio y octubre**.

Se obtuvo un estadístico de prueba de:

$$
F = 4.9412
$$

y un valor-p de:

$$
p = 0.001984
$$

Con un nivel de significancia de:

$$
\alpha = 0.05
$$

Dado que

$$
p < \alpha
$$

se **rechaza la hipótesis nula ($H_0$)**.

Por lo tanto, existe evidencia estadística suficiente para concluir que **al menos uno de los meses analizados presenta un precio promedio significativamente diferente** respecto a los demás.

Para identificar específicamente cuáles grupos difieren entre sí, se aplicó la **prueba Post-Hoc de Tukey**.

Los resultados muestran diferencias estadísticamente significativas entre:

- **Abril y Enero** ($p = 0.0013$).
- **Abril y Julio** ($p = 0.0242$).

En cambio, no se encontraron diferencias estadísticamente significativas entre los demás pares de meses, ya que sus valores-p fueron mayores que el nivel de significancia establecido ($\alpha = 0.05$).

En consecuencia, los resultados indican que **las variaciones observadas en el precio promedio de las acciones se concentran principalmente en el mes de abril**, mientras que los restantes meses presentan comportamientos estadísticamente similares.